### PR TITLE
Merged registration summary into form page

### PR DESCRIPTION
**Pull request by pferreir**

### CODE REVIEW

File: indico/htdocs/js/indico/modules/registration/form/tpls/toolbar.tpl.html [hunk 1]
```diff
@@ -25,3 +34,9 @@
             class="i-button icon-stack-minus" title="{{ 'Collapse All' | i18n }}"></a>
     </div>
 </div>
+
+<script>
+    $(document).ready(function() {
+        $('.js-dropdown').closest('.group').dropdown();
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/toolbar.tpl.html]: Shouldn't be necessary, we have this code in global.js:

```javascript
    $('.js-dropdown').each(function() {
        this.id = this.id || uniqueId();
        $(this).parent().dropdown({selector: '#' + this.id});
    });
```

File: indico/htdocs/js/indico/modules/registration/form/tpls/toolbar.tpl.html [hunk 2]
```diff
@@ -25,3 +34,9 @@
...
+    });
+</script>
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/toolbar.tpl.html]: no trailing linebreak

File: indico/modules/events/registration/__init__.py [hunk 3]
```diff
@@ -55,8 +55,10 @@ def _get_open_regforms(event):
 @template_hook('conference-home-info')
 def _inject_regform_announcement(event, **kwargs):
     regforms = _get_open_regforms(event)
+    user_has_registered = session.user and any(regform.get_registration(user=session.user) for regform in regforms)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/__init__.py]: user_has_registered = session.user and Registration.find(user=session.user).with_parent(regform).count()

Comment (ThiefMaster) [inline indico/modules/events/registration/__init__.py]: user_has_registered = session.user and Registration.find(user=session.user, is_deleted=False, Registration.registration_form.in_(regforms)).count()

Comment (ThiefMaster) [inline indico/modules/events/registration/__init__.py]: actually.. would this make sense as a util function? `user_registered_in_event(event, user)`

File: indico/modules/events/registration/controllers/display.py [hunk 4]
```diff
@@ -88,8 +82,8 @@ class RHRegistrationFormList(RHRegistrationFormDisplayBase):
 
     def _process(self):
         regforms = RegistrationForm.find_all(RegistrationForm.is_scheduled, event_id=int(self.event.id))
-        if _can_redirect_to_single_regform(regforms):
-            return redirect(url_for('.display_regform_summary', regforms[0]))
+        if len(regforms) == 1 and regforms[0].can_submit(session.user):
```
Comment (OmeGak) [inline indico/modules/events/registration/controllers/display.py]: I don't think we should prevent access to the registration summary page in any case now that it has been merged with the registration form. I don't think your changes prevent it, though, but then the redirection shouldn't be prevented either.

File: indico/modules/events/registration/templates/display/_regform_info.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                        {% trans %}Contact info{% endtrans %}
+                    </div>
+                    <div>{{ regform.contact_info }}</div>
+                </div>
+            </div>
+        {% endif %}
+    </div>
+
+    {% if regform.introduction %}
+        <div class="info-message-box">
```
Comment (OmeGak) [inline indico/modules/events/registration/templates/display/_regform_info.html]: Why not using the macro?

```jinja
{% from 'message_box.html' import message_box %}
{% call message_box('info') %}
    {{ regform.introduction }}
{% endcall %}
```

Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/_regform_info.html]: didn't you mean `{{ message_box('info', regform.introduction) }}`? ;)

Comment (OmeGak) [inline indico/modules/events/registration/templates/display/_regform_info.html]: Even better :open_hands:

Comment (pferreir) [inline indico/modules/events/registration/templates/display/_regform_info.html]: I didn't write it initially, but totally agree :+1:

File: indico/htdocs/js/indico/modules/registration/form/tpls/toolbar.tpl.html (hunk 1 above)
Comment (pferreir) [inline indico/htdocs/js/indico/modules/registration/form/tpls/toolbar.tpl.html]: Thought of it but doesn't work, as this is Angular and the element is not available on page load.

Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/toolbar.tpl.html]: Then you most likely don't need the domready around it

Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/tpls/toolbar.tpl.html]: And I'd use a more specific selector unless the dropdown js is fine with being called again after having been initialized

File: indico/modules/events/registration/__init__.py (hunk 3 above)
Comment (pferreir) [inline indico/modules/events/registration/__init__.py]: Done.

File: indico/modules/events/registration/controllers/display.py (hunk 4 above)
Comment (pferreir) [inline indico/modules/events/registration/controllers/display.py]: Yeah, this is basically useless, as the button link already points to the list or a single form according to how many there are.

**Comment by pferreir** [comment]
Merged.

**Pull request closed without merging**